core/new-gui/src/app/workspace/service/drag-drop/drag-drop.service.ts

```diff
@@ -228,6 +259,80 @@ export class DragDropService {
         y: ui.offset.top
       }
     });
+
+    if (this.suggestionOperator !== undefined) {
+      const operatorID = this.suggestionOperator.operatorID;
+      const status = true;
+      this.operatorSuggestionUnhighlightStream.next({status, operatorID});
+      this.suggestionOperator = undefined;
+    }
+  }
+
+  private handleMouseMove(): void {
+    let isDone = false;
+
+    Observable.fromEvent<MouseEvent>(window, 'mouseup').first()
+      .subscribe(
+        () => isDone = true,
+        (error) => console.error(error)
+      );
+
+    Observable.fromEvent<MouseEvent>(window, 'mousemove').auditTime(100).map(
+      value => [value.clientX, value.clientY]
+    ).subscribe(
+      (value) => {
+        if (!isDone) {
+          this.mouseAt = {x: value[0], y: value[1]};
+          this.findClosestOperator();
+        }
+      },
+      (error) => console.error(error)
+    );
   }
 
+  private findClosestOperator(): void {
+    const operator_list = this.workflowActionService.getTexeraGraph().getAllOperators();
+    const distance: number[] = [Number.MAX_VALUE]; // keep tracking the closest operator
+
+    for (let i = 0; i < operator_list.length; i++) {
+
+      // Get an operator position details by using getJointOperatorCellPostion in workflowActionService
+      const position = this.workflowActionService.getJointGraphWrapper().getJointOperatorCellPostion(operator_list[i].operatorID);
+      if (position !== undefined && this.mouseAt !== undefined) {
+        // calculate the distance between the mouse and the operator
+        const dis = (this.mouseAt.x - position[0]) ** 2 + (this.mouseAt.y - position[1]) ** 2;
```

Comment:
you need `sqrt` the `dis` variable here to correctly calculate the distance

Reply:
I thought the results will be the same. Is it necessary to use the sqrt?

Reply:
This doesn't need sqrt for distance compare.